### PR TITLE
Fixing live tests for UploadAsync_ReadOnlyError

### PR DESCRIPTION
**Pull request by amnguye**
After this PR https://github.com/Azure/azure-sdk-for-net/pull/13592 was pushed with the added test, the test I added failed even though it passed CI. I'm trying to fix it with this right now.

### CODE REVIEW

**Review by kasobol-msft — APPROVED** [review]

**Comment by pakrym** [comment]
There are two unrelated issues here:
1. A bug in the new HttpWebRequest transport that's being fixed by https://github.com/Azure/azure-sdk-for-net/pull/13750
2. Usage of instrumented client in `ShareFileClient readOnlyClient = new ShareFileClient(new Uri(sasUri.ToString()));` that tries to access a live resource even in a playback mode.

**Review by seanmcc-msft — APPROVED** [review]

File: sdk/storage/Azure.Storage.Files.Shares/tests/FileClientTests.cs [hunk 1]
```diff
@@ -2006,33 +2004,33 @@ public async Task UploadAsync_ReadOnlyError()
...
             {
-                Query = sasBuilder.ToSasQueryParameters(
-                    new StorageSharedKeyCredential(
+                Query = sasBuilder.ToSasQueryParameters(new StorageSharedKeyCredential(
                         TestConfigDefault.AccountName,
                         TestConfigDefault.AccountKey)).ToString()
             };
 
-            ShareFileClient readOnlyClient = new ShareFileClient(new Uri(sasUri.ToString()));
+            ShareFileClient readOnlyClient = new ShareFileClient(new Uri(sasUri.ToString()), GetOptions());
```
Comment (jaschrep-msft) [inline sdk/storage/Azure.Storage.Files.Shares/tests/FileClientTests.cs:2025]: Should this be instrumented?

Comment (amnguye) [reply sdk/storage/Azure.Storage.Files.Shares/tests/FileClientTests.cs:2025]: `GetOptions()` returns a `InstrumentClientOptions`, so it should be instrumented.

Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Files.Shares/tests/FileClientTests.cs:2025]: @tg-msft has told me before that we also need to write the new Clients in `InstrumentClient()`.

**Pull request merged**